proxyrunReq_test.go

```diff
@@ -0,0 +1,229 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type mockResponseWriterWithCode struct {
+	http.ResponseWriter
+	statusCode int
+}
+
+// TestRunQueryFail function statResponseWriter's statusCode will not be 200, but the query has been proxied
+// The request will be retried 1 time with a different host in the same replica
+func TestRunQueryFail(t *testing.T) {
```

Comment:
What about calling it
```suggestion
func TestQueryWithRetryFail(t *testing.T) {
```

Reply:
Done